utils/release/lock-master.mjs

```diff
@@ -0,0 +1,32 @@
+import {Octokit} from 'octokit';
+
+const REPO_OWNER = 'coveo';
+const REPO_NAME = 'cli';
+
+export const limitWriteAccessToBot = () =>
+  changeBranchRestrictions({
+    users: [],
+    apps: ['developer-experience-bot'],
+    teams: [],
+  });
+
+export const removeWriteAccessRestrictions = () =>
+  changeBranchRestrictions(null);
+
+async function changeBranchRestrictions(restrictions) {
+  const octokit = new Octokit({auth: process.env.GITHUB_CREDENTIALS});
+
+  const currentProtection = await octokit.rest.repos.getBranchProtection({
+    owner: REPO_OWNER,
+    repo: REPO_NAME,
+    branch: 'master',
+  });
```

Comment:
With the typescript language server, it looks like `getBranchProtection` doesn't directly return the protections, it returns a `data` object. If so, it looks like you have an error.

```suggestion
  const {status, data: currentProtection} =
    await octokit.rest.repos.getBranchProtection({
      owner: REPO_OWNER,
      repo: REPO_NAME,
      branch: 'master',
    });
  if (status !== 200) {
    throw 'Could not get current branch protections.';
  }
```

Reply:
Good point for the data obj.
We don't need any handling tho, octokit throw when a request is nok